apps/Testkit.java/Testkit.java.android/Tests/AndroidClient2/app/src/main/java/com/couchbase/androidclient/MainActivity.java

```diff
@@ -0,0 +1,121 @@
+package com.couchbase.androidclient;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import com.couchbase.lite.Database;
+import com.couchbase.lite.DatabaseChange;
+import com.couchbase.lite.DatabaseChangeListener;
+import com.couchbase.lite.DatabaseConfiguration;
+import com.couchbase.lite.Document;
+import com.couchbase.lite.Log;
+import com.couchbase.lite.Replicator;
+import com.couchbase.lite.ReplicatorConfiguration;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Random;
+import java.util.TimerTask;
+
+public class MainActivity extends AppCompatActivity {
+
+  private Database database;
+  private Document doc;
+  private Replicator replicator;
+  private int numOfDocs = 4;
+  private long scenarioRunTimeMinutes = 1;
+  private String syncGatewayURL = "blip://192.168.33.11:4984/db/";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    numOfDocs = getIntent().getIntExtra("numOfDocs", 4);
+    scenarioRunTimeMinutes = getIntent().getLongExtra("scenarioRunTimeMinutes", 1);
+    syncGatewayURL = getIntent().getStringExtra("syncGatewayURL");
```

Comment:
Removed default values, getIntExtra forcing to use default values, so changed to 0 . If the values for numOfDocs and scnarioRunTimeMinutes are 0, test won't run, it stops